beefy-gadget/src/lib.rs

```diff
@@ -111,40 +111,23 @@ pub async fn start_beefy_gadget<Block, Pair, Backend, Client, Network, SyncOracl
 	);
 
 	let at = BlockId::hash(client.info().best_hash);
+
 	let validator_set = client
 		.runtime_api()
 		.validator_set(&at)
-		.expect("Failed to get BEEFY authorities");
-
-	let local_id = match validator_set
-		.validators
-		.iter()
-		.find(|id| SyncCryptoStore::has_keys(&*key_store, &[(id.to_raw_vec(), KEY_TYPE)]))
-	{
-		Some(id) => {
-			info!(target: "beefy", "🥩 Starting BEEFY worker with local id: {:?}", id);
-			worker::BeefyId::Validator(id.clone())
-		}
-		None => {
-			info!(target: "beefy", "🥩 No local id found, BEEFY worker will be gossip only.");
-			worker::BeefyId::None
-		}
-	};
+		.expect("Failed to get BEEFY validator set");
```

Comment:
So this obviously works for now, but what if the client is not fully in sync with the network? I think we need a bit more sophisticated mechanism, where we actually update the validator set id when new blocks are imported / we get synced up.

Reply:
Also regarding `expect` - the assumption here is that the beefy pallet is part of the runtime (pretty much since genesis), however existing changes will most likely upgrade to beefy pallet, so I think we should be handling this error a bit more gracefuly.

Could you log issues to make sure this is covered?

Reply:
For the out-of-sync case, please see #112 

>  however existing changes will most likely upgrade to beefy pallet,

I think I don't get, what you mean by this. Could you please elaborate?

Reply:
> I think I don't get, what you mean by this. Could you please elaborate?

What I mean is that BEEFY will be introduced as a runtime upgrade to chains like Kusama, Polkadot and maybe even Rococo.
This means that the node (and the gadget code) needs to handle a case where the pallet is not present gracefuly. The client will be upgraded first and after that will the runtime get the pallet. This means that the gadget needs to be able to enable itself when it detect that the pallet becomes active in the runtime.

Reply:
Thanks for the clarification! Filed #113 for this.